### PR TITLE
Fix a few bugs detected by FindBugs

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

**Comment by eregon** [comment]
Here are the current remaning warnings:

```
H C BIT: Bitwise OR of signed byte value computed in org.truffleruby.algorithms.SipHash.hash24(long, long, byte[], int, int)   At SipHash.java:[line 130]
L B ES: Comparison of String objects using == or != in org.truffleruby.parser.ast.OpElementAsgnParseNode.isOr()   At OpElementAsgnParseNode.java:[line 105]
L B ES: Comparison of String objects using == or != in org.truffleruby.parser.ast.OpElementAsgnParseNode.isAnd()   At OpElementAsgnParseNode.java:[line 109]
L B ES: Comparison of String objects using == or != in org.truffleruby.parser.scope.StaticScope.namesAreInterned(String[])   At StaticScope.java:[line 131]
M M DC: Possible doublecheck on org.truffleruby.RubyContext.consoleHolder in org.truffleruby.RubyContext.getConsoleHolder()  At RubyContext.java:[lines 366-368]
L B ES: Comparison of String objects using == or != in org.truffleruby.core.cast.NameToJavaStringNodeGen.executeToJavaString(VirtualFrame, Object)   At NameToJavaStringNodeGen.java:[line 79]
L B ES: Comparison of String objects using == or != in org.truffleruby.core.cast.NameToJavaStringNodeGen.execute(VirtualFrame)   At NameToJavaStringNodeGen.java:[line 140]
L B ES: Comparison of String objects using == or != in org.truffleruby.core.cast.NameToJavaStringNodeGen.executeAndSpecialize(VirtualFrame, Object)   At NameToJavaStringNodeGen.java:[line 260]
M B ES: Comparison of String objects using == or != in org.truffleruby.language.SnippetNode.ensureConstantExpressionParameters(String, Object[])   At SnippetNode.java:[line 102]
L B ES: Comparison of String objects using == or != in org.truffleruby.core.binding.BindingNodesFactory$LocalVariableSetNodeFactory$LocalVariableSetNodeGen.execute(VirtualFrame)   At BindingNodesFactory.java:[line 568]
L B ES: Comparison of String objects using == or != in org.truffleruby.core.binding.BindingNodesFactory$LocalVariableSetNodeFactory$LocalVariableSetNodeGen.executeAndSpecialize(Object, Object, Object)   At BindingNodesFactory.java:[line 614]
H B ES: Comparison of String parameter using == or != in org.truffleruby.language.constants.LookupConstantWithDynamicScopeNode.lookupConstant(VirtualFrame, Object, String)   At LookupConstantWithDynamicScopeNode.java:[line 39]
H B ES: Comparison of String parameter using == or != in org.truffleruby.language.constants.LookupConstantWithLexicalScopeNode.lookupConstant(VirtualFrame, Object, String)   At LookupConstantWithLexicalScopeNode.java:[line 45]
H B ES: Comparison of String parameter using == or != in org.truffleruby.parser.parser.ParserSupport.arg_var(String)   At ParserSupport.java:[line 1348]
H B ES: Comparison of String parameter using == or != in org.truffleruby.parser.parser.ParserSupport.shadowing_lvar(String)   At ParserSupport.java:[line 1366]
L B ES: Comparison of String objects using == or != in org.truffleruby.core.binding.BindingNodesFactory$LocalVariableGetNodeFactory$LocalVariableGetNodeGen.execute(VirtualFrame)   At BindingNodesFactory.java:[line 353]
L B ES: Comparison of String objects using == or != in org.truffleruby.core.binding.BindingNodesFactory$LocalVariableGetNodeFactory$LocalVariableGetNodeGen.executeAndSpecialize(Object, Object)   At BindingNodesFactory.java:[line 399]
H B ES: Comparison of String parameter using == or != in org.truffleruby.language.methods.LookupMethodNodeGen.executeLookupMethod(VirtualFrame, Object, String)   At LookupMethodNodeGen.java:[line 46]
L B ES: Comparison of String objects using == or != in org.truffleruby.language.methods.LookupMethodNodeGen.execute(VirtualFrame)   At LookupMethodNodeGen.java:[line 76]
L B ES: Comparison of String objects using == or != in org.truffleruby.language.methods.LookupMethodNodeGen.executeAndSpecialize(VirtualFrame, Object, Object)   At LookupMethodNodeGen.java:[line 108]
H B IMSE: Dubious catching of IllegalMonitorStateException in org.truffleruby.core.mutex.MutexOperations.unlock(ReentrantLock, DynamicObject, RubyNode)  At MutexOperations.java:[line 97]
H D UCF: Useless control flow to next line in org.truffleruby.core.string.StringNodes$ScrubNode.scrubAscciIncompatible(VirtualFrame, DynamicObject, DynamicObject)  At StringNodes.java:[line 1442]
M D IM: Check for oddness that won't work for negative numbers in org.truffleruby.stdlib.bigdecimal.BigDecimalNodes$SqrtNode.bigSqrt(BigDecimal, MathContext)  At BigDecimalNodes.java:[line 833]
```

File: src/main/java/org/truffleruby/parser/parser/SuppressFBWarnings.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+@Retention(RetentionPolicy.CLASS)
+@interface SuppressFBWarnings {
+    /**
+     * The set of FindBugs
+     * <a href="http://findbugs.sourceforge.net/bugDescriptions.html">warnings</a> that are to be
+     * suppressed in annotated element. The value can be a bug category, kind or pattern.
+     */
+    String[] value();
+}
```
Comment (chrisseaton) [inline src/main/java/org/truffleruby/parser/parser/SuppressFBWarnings.java]: I don't get it - how can FindBugs look for our interface?

Comment (eregon) [inline src/main/java/org/truffleruby/parser/parser/SuppressFBWarnings.java]: AFAIK it just look at the name of the interface, regardless of the package.
This has the advantage to not need to import FindBugs as a dependency.

**Comment by eregon** [comment]
@pitr-ch Could you look at:
```
M D IM: Check for oddness that won't work for negative numbers in org.truffleruby.stdlib.bigdecimal.BigDecimalNodes$SqrtNode.bigSqrt(BigDecimal, MathContext)  At BigDecimalNodes.java:[line 833]
```

@nirvdrum Could you look at:
```
H D UCF: Useless control flow to next line in org.truffleruby.core.string.StringNodes$ScrubNode.scrubAscciIncompatible(VirtualFrame, DynamicObject, DynamicObject)  At StringNodes.java:[line 1442]
```

@chrisseaton Could you look at:
```
H C BIT: Bitwise OR of signed byte value computed in org.truffleruby.algorithms.SipHash.hash24(long, long, byte[], int, int)   At SipHash.java:[line 130]
```

http://findbugs.sourceforge.net/bugDescriptions.html for details.

**Pull request merged**